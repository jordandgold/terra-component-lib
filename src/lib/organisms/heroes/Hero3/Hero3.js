import React from "react";
import "./Hero3.scss";
import PropTypes from "prop-types";

const Hero3 = props => {
  const { images, headerText, subHeaderText } = props;
  return (
    <section className="ter-hero-3">
      <figure className="ter-hero-3__hero-image ter-hero-3__hero-image--4x1">
        <img
          src={images.imageFourByOne.url}
          alt={images.imageFourByOne.altText}
        />
      </figure>
      <figure className="ter-hero-3__hero-image ter-hero-3__hero-image--2x1">
        <img
          src={images.imageTwoByOne.url}
          alt={images.imageTwoByOne.altText}
        />
      </figure>
      <div className="ter-hero-3__container">
        <aside className="ter-hero-3__card ter-card">
          <div className="ter-card__body">
            <h1 className="ter-hero-3__header">{headerText}</h1>
            <div className="ter-hero-3__sub-header">{subHeaderText}</div>
          </div>
        </aside>
      </div>
    </section>
  );
};

export default Hero3;
