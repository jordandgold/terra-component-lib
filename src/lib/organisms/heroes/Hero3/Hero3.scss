@import "../../../global-styles/terra.scss";

.ter-hero-3 {
  &__hero-image {
    width: 100%;
    height: auto;
    margin: 0;
    padding: 0;
    img {
      width: 100%;
      height: auto;
      display: block;
    }
    &--4x1 {
      display: none;
      @media screen and (min-width: $breakpoint-sm) {
        display: block;
      }
    }
    &--2x1 {
      display: block;
      @media screen and (min-width: $breakpoint-sm) {
        display: none;
      }
    }
  }
  &__container {
    @include container-wrap;
    display: flex;
    width: 100%;
    height: auto;
    transform: translate(0, calc(#{spacing("xxl")} * -1));
    justify-content: center;
    @media screen and (min-width: $breakpoint-xs) {
      transform: translate(0, calc(#{spacing("super")} * -1));
    }
  }
  &__card {
    background-color: rgba(255, 255, 255, 0.95);
    display: block;
    flex-basis: 550px;
    margin-top: auto;
    margin-bottom: calc(#{spacing("md")} * -1);
    align-self: center;
    @media screen and (min-width: $breakpoint-xs) {
      margin-bottom: calc(#{spacing("sm")} * -1);
    }
  }
  &__header {
    font-size: font-size("xl");
    text-align: center;
  }
  &__sub-header {
    font-size: font-size("md");
    font-weight: font-weight("light");
    margin-bottom: spacing("md");
    text-align: center;
  }
}
