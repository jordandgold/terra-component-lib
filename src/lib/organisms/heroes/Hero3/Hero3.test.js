import React from "react";
import { shallow } from "enzyme";
import Hero3 from "./Hero3";

describe("Hero4", () => {
  let wrapper;
  const mockImages = {
    imageFourByOne: {
      url: "http://fpoimg.com/1600x400?text=4:1",
      altText: "placeholder image"
    },
    imageTwoByOne: {
      url: "http://fpoimg.com/1600x800?text=2:1",
      altText: "placeholder image"
    }
  };

  beforeEach(() => {
    wrapper = shallow(
      <Hero3
        images={mockImages}
        headerText="Good design is innovative."
        subHeaderText="Esse aliquip ad in et ut ipsum paEu elit consectetur aliquip excepteur fugiat ut qui dolor pariatur consectetur.riatur elit quis."
      />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
