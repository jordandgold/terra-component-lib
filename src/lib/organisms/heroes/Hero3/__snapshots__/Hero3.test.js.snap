// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero4 should match the snapshot 1`] = `
<section
  className="ter-hero-3"
>
  <figure
    className="ter-hero-3__hero-image ter-hero-3__hero-image--4x1"
  >
    <img
      alt="placeholder image"
      src="http://fpoimg.com/1600x400?text=4:1"
    />
  </figure>
  <figure
    className="ter-hero-3__hero-image ter-hero-3__hero-image--2x1"
  >
    <img
      alt="placeholder image"
      src="http://fpoimg.com/1600x800?text=2:1"
    />
  </figure>
  <div
    className="ter-hero-3__container"
  >
    <aside
      className="ter-hero-3__card ter-card"
    >
      <div
        className="ter-card__body"
      >
        <h1
          className="ter-hero-3__header"
        >
          Good design is innovative.
        </h1>
        <div
          className="ter-hero-3__sub-header"
        >
          Esse aliquip ad in et ut ipsum paEu elit consectetur aliquip excepteur fugiat ut qui dolor pariatur consectetur.riatur elit quis.
        </div>
      </div>
    </aside>
  </div>
</section>
`;
